docs/architecture-decisions/2024-03-06_local_dev_ui_packaging.md

```diff
@@ -0,0 +1,110 @@
+- Link to PoC: [Electron PoC](https://github.com/negar-abbasi/electron-poc)
+
+### Option 2 - Tauri
+
+[Tauri](https://tauri.app/about/intro) is a toolkit that helps developers make applications for the major desktop platforms - using virtually any frontend framework in existence. The core is built with Rust, and the CLI leverages Node.js making Tauri a genuinely polyglot approach to creating and maintaining great apps.
+
+**Pros**
+Tauri supports all intended operations for the local dev UI MVP via their JavaScript API without the need to write any Rust.
+
+- It can manage the [file systems](https://tauri.app/v1/api/js/fs), launch another [process](https://tauri.app/v1/api/js/process), run a [shell command](https://tauri.app/v1/api/js/shell).
+- Tauri integrates well with major web frameworks. [`create-tauri-app`](https://github.com/tauri-apps/create-tauri-app) can bootstrap a working app within minutes.
+  - Once bootstrapped, the web app can be bundled individually and deployed as a website. Below is the `npm script` Tauri generates for a Vite project, we can see that it supports `vite build`
+  ```json
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview",
+    "tauri": "tauri"
+  },
+  ```
+- For [compiling binaries](https://tauri.app/v1/guides/building/), the Tauri Bundler supports
+  - Windows: setup.exe, .msi
+  - macOS: .app, .dmg
+  - Linux: .deb, .appimage
+- The Tauri Bundler supports code signing for:
+  - [Windows](https://tauri.app/v1/guides/distribution/sign-windows)
+  - [Linux](https://tauri.app/v1/guides/distribution/sign-linux)
+  - [macOS](https://tauri.app/v1/guides/distribution/sign-macos)
+- Tauri offers a [built-in updater](https://tauri.app/v1/guides/distribution/updater) for the NSIS (Windows), MSI (Windows), AppImage (Linux) and App bundle (macOS) distribution formats.
+- Tauri claims to be less resource hungry than Electron, however, I haven't been able to test it intensively in this work.
+
+**Cons**
+
+- If we need to extend the functionalities beyond the support of Tauri's JavaScript API, we will need to write the code in Rust, which would be a new language in the AlgoKit ecosystem and a less common skill in market.
+- Arm based Linux runner is required to build binaries for Arm Linux. Currently, GitHub doesn't support Arm based runner yet, but will be [soon](https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/) in the future.
+- At the point of writing, building with `snap` (for Linux) isn't officially supported by Tauri. There is a open [PR](https://github.com/tauri-apps/tauri/pull/6532).
+- Tauri relies on [Webview](https://tauri.app/v1/references/webview-versions/) which are not the same across platforms. It introduce a risk that we will need to make CSS works across different Webview versions.
+  - This can be a none issue because we will need to make our CSS works across modern browsers for the website.
```

Comment:
Worth mentioned, that we aren't actually planning on deploying an explorer site. We just want to have that option.

Reply:
I remove this sentence,